app/models/game.rb

```diff
@@ -29,29 +29,210 @@ def populate_board!
+    @remaining_opponent_pieces.delete_if { |remaining_opponent_piece| remaining_opponent_piece.type.eql? 'King' }
+  
+  # Store the coordinate sets between the player piece causing check and the opponent king
+    in_between_coordinates = set_coordinates_between_piece_causing_check_and_opponent_king
+    
+  # Iterate through coordinate sets between piece causing check, and opponent king
+    in_between_coordinates.each do |x_of_threat, y_of_threat|
+      @remaining_opponent_pieces.each do |opponent_piece|
+      # Check if it would be valid for the opponent_piece move to the specified
+      # coordinate set between the piece causing check and opponent king
+        if opponent_piece.valid_move? x_of_threat, y_of_threat
+        # Check if the opponent_piece wouldn't obstructed while attempting to move
+        # to the specified coordinate set between the piece causing check 
+          unless opponent_piece.is_obstructed? x_of_threat, y_of_threat
+          # If both criteria are met, at least one opponent_piece can block the piece causing check
+            return true
+          end
+        end
+      end
+    end
+    
+  # Return false: no opponent_piece can block the player piece causing check
     false
   end
+  
+  def checked_king_can_escape?
+  # Store opponent_king coordinates locally
+    okx = @okx
+    oky = @oky
+    
+  # Build two-dimensional array of opponent king's surrounding coordinate sets
+  # (leave behind any coordinate sets that contain a 0 or 9 (would be invalid moves))
+    possible_escape_coordinates_for_king = [
+      [okx, oky - 1],
+      [okx, oky + 1],
+      [okx + 1, oky],
+      [okx - 1, oky],
+      [okx - 1, oky - 1],
+      [okx - 1, oky + 1],
+      [okx + 1, oky - 1],
+      [okx + 1, oky + 1]
+    ].delete_if { |coordinate_set| coordinate_set.include? 0 }
+     .delete_if { |coordinate_set| coordinate_set.include? 9 }
+    
+  # Iterate through opponent king's valid surrounding coordinate sets
+    possible_escape_coordinates_for_king.each do |escape_x, escape_y|
+    # Check if it would be valid for the opponent_king to move to the specified surrounding coordinate set
+      if @opponent_king.valid_move? escape_x, escape_y
+      # Check if the opponent_king wouldn't obstructed while attempting to move to the specified surrounding coordinate set
+        unless @opponent_king.is_obstructed? escape_x, escape_y
+        # Check if a player piece doesn't occupy the specified surrounding coordinate set
+          unless @opponent_king.player_piece_occupies_destination? escape_x, escape_y
+          # Check if the opponent_king wouldn't enter check while attempting to move to the specified surroudning coordinate set
+            unless @opponent_king.move_would_cause_check? escape_x, escape_y
+            # If above criteria are met, the opponent_king can escape check
+              return true
+            end
+          end
+        end
+      end
```

Comment:
You can save even more space and reduce the additional levels of block nesting if you used 'next if' like so:
```
if @opponent_king.valid_move? escape_x, escape_y
  next if @opponent_king.is_obstructed? escape_x, escape_y
  next if @opponent_king.player_piece_occupies_destination? escape_x, escape_y
  next if @opponent_king.move_would_cause_check? escape_x, escape_y
  return true
end
```